pkg/requestmeta/requestmeta.go

```diff
@@ -0,0 +1,39 @@
+package requestmeta
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+// RequestMetadataHeaderKey defines a key in the request metadata header.
+type RequestMetadataHeaderKey string
+
+// BoolRequestMetadataHeaderKey defines a key for a boolean vlaue in the request metadata header.
+type BoolRequestMetadataHeaderKey RequestMetadataHeaderKey
+
+const (
+	// RequestServerVersion, if specified in a request header, asks SpiceDB to return its
+	// server version in the response header (if supported).
+	// Value: `1`
+	RequestServerVersion BoolRequestMetadataHeaderKey = "io.spicedb.requestversion"
+)
+
+// AddRequestHeaders returns a new context with the given values as request headers.
+func AddRequestHeaders(ctx context.Context, keys ...BoolRequestMetadataHeaderKey) context.Context {
```

Comment:
Shouldn't this be `AddBooleanRequestHeaders`? I imagine there will be headers in the future that have contents.

Reply:
The idea is that you can Add boolean headers, but Set non-bool headers (since they have values)

Reply:
ah, yeah that makes sense